account_invoice_production_lot/invoice.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Author: Lorenzo Battistini <lorenzo.battistini@agilebg.com>
+#    Copyright (C) 2011 Domsense s.r.l. (<http://www.domsense.com>).
+#    Copyright (C) 2013 Agile Business Group sagl (<http://www.agilebg.com>)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+
+
+class account_invoice_line(models.Model):
+
+    @api.one
+    def _get_prod_lots(self):
+        if not self.move_line_ids and not self.order_lines:
+            return
+        if self.move_line_ids:
```

Comment:
You can do `self.mapped('move_line_ids.lot_ids')` for a simpler handling of this